docs/content/ActionList.mdx

```diff
@@ -196,29 +196,29 @@ const SelectFields = () => {
   }
 
   return (
-    <ActionList selectionVariant="multiple">
-      <ActionList.Group title="Visible fields">
+    <>
+      <ActionList selectionVariant="multiple" headingProps={{title: 'Visible fields'}}>
         {visibleOptions.map(option => (
           <ActionList.Item key={option.text} selected={true} onSelect={() => toggle(option.text)}>
             {option.text}
           </ActionList.Item>
         ))}
-      </ActionList.Group>
-      <ActionList.Group
-        title="Hidden fields"
+      </ActionList>
+      <ActionList
         selectionVariant={
           /** selectionVariant override on Group: disable selection if there are no options */
           hiddenOptions.length ? 'multiple' : false
         }
+        headingProps={{title: 'Hidden fields'}}
```

Comment:
That would be ideal, but this doesn't let us assign `aria-labelledby` correctly.

As far as I understand, we don't want users to add more than one heading, and we need to be able to use a generated id for the heading to label the list itself. Rendering the heading with a slot doesn't help with that, as we can't get the id out of the heading.

Also, we specifically don't want users to pass arbitrary tags to `as`, hence the `headingLevel` prop, as this can replicate inaccessible patterns.

If there's a better way to implement this with those requirements in mind, I'm happy to change this. It does feel a little clunky.

Reply:
> and we need to be able to use a generated id for the heading to label the list itself. Rendering the heading with a slot doesn't help with that, as we can't get the id out of the heading.

Can you tell me a little more about that?

Do we want to keep the id added by the developer or can we always use a generated one?

We do the second with slots in ActionList.Item for descriptions: https://github.com/primer/react/blob/main/src/ActionList/Item.tsx#L173

> Also, we specifically don't want users to pass arbitrary tags to as, hence the headingLevel prop, as this can replicate inaccessible patterns.

We restrict the allowed values for `as` prop :) https://github.com/primer/react/blob/main/src/Heading/Heading.tsx#L10

Reply:
I don't think we should expect developers to add an id explicitly. In this case, the id probably isn't meaningful except for ensuring that the id can be used to link via the `aria-labelledby` attribute. I don't think slots help us there since rendered components are "sealed". We can't get the id out.

Restricting the `as` values will work though. Thanks for the reference.